### PR TITLE
Ensure payment source persistence

### PR DESCRIPTION
**Pull request by caudatecoder**
- Make Spree::CreditCard `acts_as_paranoid`
- Add payment source presence validation

### CODE REVIEW

**Comment by caudatecoder** [comment]
@VygovskySergey thanks, good catch

Still, I think it's better to add the validation and check there for `source_required?` && `presence`

**Comment by damianlegawiec** [comment]
@VygovskySergey this `source_required?` is kinda whacky - we should enforce AR validation for this - it results later with payment records without source because someone removed it accidentally or sth. It should be protected.

**Comment by VygovskySergey** [comment]
@damianlegawiec For example right now in `Afterpay` there is no source, that is why it is crucial, to have option for payment methods like this. But I agree that with checking `source_required? == true` it should be validated.

**Review by damianlegawiec — APPROVED** [review]

**Review by btolarz — APPROVED** [review]

File: core/app/models/spree/credit_card.rb [hunk 1]
```diff
@@ -2,6 +2,12 @@ module Spree
   class CreditCard < Spree::Base
     include ActiveMerchant::Billing::CreditCardMethods
 
+    if !ENV['DISABLE_DB_CONNECTION'] &&
```
Comment (damianlegawiec) [inline core/app/models/spree/credit_card.rb]: @caudatecoder can we rename it to `SPREE_DISABLE_DB_CONNECTION` ? thanks :)

Comment (damianlegawiec) [reply core/app/models/spree/credit_card.rb]: @caudatecoder it's the first instance of this ENV in the codebase

Comment (caudatecoder) [reply core/app/models/spree/credit_card.rb]: @damianlegawiec oh, ok, it was in an unmerged PR :)
Will do then

**Pull request merged**